### PR TITLE
Capabilities registry OCR3 support

### PR DESCRIPTION
**Pull request by george-dorin**

### CODE REVIEW

File: pkg/loop/internal/capabilities_registry.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+	return reply, nil
+}
+
+func (c *capabilitiesRegistryServer) Add(ctx context.Context, request *pb.AddRequest) (*emptypb.Empty, error) {
+	conn, err := c.Dial(request.CapabilityID)
+	if err != nil {
+		return &emptypb.Empty{}, ErrConnDial{Name: "Add", ID: request.CapabilityID, Err: err}
+	}
+	client := newBaseCapabilityClient(c.BrokerExt, conn)
```
Comment (cedric-cordenier) [inline pkg/loop/internal/capabilities_registry.go:304]: You need to check the execution type and wrap it in the correct underlying client type.

File: pkg/loop/internal/capabilities_registry.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+func (c *capabilitiesRegistryServer) Get(ctx context.Context, request *pb.GetRequest) (*pb.GetReply, error) {
+	capability, err := c.impl.Get(ctx, request.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	var cRes Resource
+	id, cRes, err := c.ServeNew("someCapability", func(s *grpc.Server) {
+		pb.RegisterBaseCapabilityServer(s, &baseCapabilityServer{impl: capability})
```
Comment (cedric-cordenier) [inline pkg/loop/internal/capabilities_registry.go:179]: When you get a capability, you need to wrap it in the correct underlying client type.

File: pkg/loop/internal/capabilities_registry.go [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	reply := &pb.ListReply{}
+
+	var resources []Resource
+	for _, cap := range capabilities {
+		id, res, err := c.ServeNew("List", func(s *grpc.Server) {
+			pb.RegisterBaseCapabilityServer(s, &baseCapabilityServer{impl: cap})
```
Comment (cedric-cordenier) [inline pkg/loop/internal/capabilities_registry.go:286]: Wrap in correct underlying type.

File: pkg/loop/internal/capabilities_registry.go [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+		clients = append(clients, client)
+	}
+
+	return clients, nil
+}
+
+func (cr *capabilitiesRegistryClient) Add(ctx context.Context, c capabilities.BaseCapability) error {
+	var cRes Resource
+	id, cRes, err := cr.ServeNew("someCapability", func(s *grpc.Server) {
+		pb.RegisterBaseCapabilityServer(s, &baseCapabilityServer{impl: c})
```
Comment (cedric-cordenier) [inline pkg/loop/internal/capabilities_registry.go:133]: Same thing: this needs to be the right type.

**Review by cedric-cordenier — APPROVED** [review]

File: pkg/loop/internal/test/capabilities_registry.go
Comment (jmank88) [inline pkg/loop/internal/test/capabilities_registry.go:1]: Will we have test coverage for this type later?

**Review by cedric-cordenier — APPROVED** [review]

**Pull request merged**